pkg/common/diskutil/file_test.go

```diff
@@ -0,0 +1,55 @@
+package diskutil
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	tests := []struct {
+		name string
+		data []byte
+		mode os.FileMode
+	}{
+		{
+			name: "basic test",
+			data: []byte("Hello, World"),
+			mode: 0600,
+		},
+		{
+			name: "empty",
+			data: []byte{},
+			mode: 0440,
+		},
+		{
+			name: "binary",
```

Comment:
nit: it's isn't clear to me what benefit this test case brings to the suite (since there is no special case handling of the byte slice passed to AtomicWriteFile)

Reply:
I was thinking one /potential way/ you could screw up a file writing function was by doing some kind of conversion to a string that's potentially lossy.  Maybe that's more a concern in something like Python and we don't need the test here, but it seems like it shouldn't really hurt to have it.